mtags/src/main/scala-2/scala/meta/internal/pc/SignatureHelpProvider.scala

```diff
@@ -487,12 +487,30 @@ class SignatureHelpProvider(val compiler: MetalsGlobal)(implicit
           if (isActiveSignature) t.call.qualTpe
           else alternativeTpe
 
+        // Check if this is a case class constructor pattern by verifying that the unapply method
+        // belongs to the companion object of the case class result type
+        val isCaseClassConstructor =
+          t.call.isUnapplyMethod && tpe.resultType.typeSymbol.isCaseClass &&
+            method.owner.isModuleClass && method.owner.companionClass == tpe.resultType.typeSymbol
+
         val paramss: List[List[Symbol]] =
           if (!isActiveSignature) {
-            mparamss(tpe, t.call.isUnapplyMethod)
+            if (isCaseClassConstructor) {
+              val constructorParamss =
+                tpe.resultType.typeSymbol.primaryConstructor.paramLists
+              constructorParamss
```

Comment:
We should probably zip the names from these with mparamss(tpe, t.call.isUnapplyMethod) so that we get instatntiated types.

When using the case class symbol we get the base types for generic, which might be useful in some cases like with Any, but not when the type is actually instatiated